src/v/model/metadata.h

```diff
@@ -234,6 +234,7 @@ struct topic_namespace_view {
 };
 
 struct topic_namespace {
+    topic_namespace() = default;
```

Comment:
now can a topic_namespace make sense without a topic or a namespace. maybe define intermediary types.

Reply:
got rid of it, using `std::optional` for `topic_manifest`